internal/resource/config/oauth/client/oauth_client_data_source.go

```diff
@@ -0,0 +1,764 @@
+package oauthclient
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/datasource/common/id"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/datasource/common/resourcelink"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/configvalidators"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &oauthClientDataSource{}
+	_ datasource.DataSourceWithConfigure = &oauthClientDataSource{}
+)
+
+var (
+	secondarySecretsDataSourceAttrType = map[string]attr.Type{
+		"encrypted_secret": basetypes.StringType{},
+		"expiry_time":      basetypes.StringType{},
+	}
+
+	clientAuthDataSourceAttrType = map[string]attr.Type{
+		"type":                                  basetypes.StringType{},
+		"encrypted_secret":                      basetypes.StringType{},
+		"secondary_secrets":                     basetypes.SetType{ElemType: basetypes.ObjectType{AttrTypes: secondarySecretsDataSourceAttrType}},
+		"client_cert_issuer_dn":                 basetypes.StringType{},
+		"client_cert_subject_dn":                basetypes.StringType{},
+		"enforce_replay_prevention":             basetypes.BoolType{},
+		"token_endpoint_auth_signing_algorithm": basetypes.StringType{},
+	}
+)
+
+// Create a Administrative Account data source
+func NewOauthClientDataSource() datasource.DataSource {
+	return &oauthClientDataSource{}
+}
+
+// oauthClientDataSource is the datasource implementation.
+type oauthClientDataSource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type oauthClientDataSourceModel struct {
```

Comment:
If this is the same as the resource model, we could just use that instead of redeclaring it here

Reply:
changed name to `oauthClientModel` and referenced in both resource and data source.